app/src/lib/stores/pull-request-store.ts

```diff
@@ -249,6 +253,28 @@ export class PullRequestStore {
     return result.reverse()
   }
 
+  /**
+   * Gets all records in the Pull Request database for the given repository
+   * and its parent/upstream, if it has one.
+   */
+  private async getAllRecords(
+    repository: GitHubRepository
+  ): Promise<ReadonlyArray<IPullRequest>> {
+    if (repository.dbID === null) {
+      return []
+    }
+    if (repository.fork && repository.parent && repository.parent.dbID) {
+      const [upstreamRecords, forkRecords] = await Promise.all([
+        this.db.getAllPullRequestsInRepository(repository.parent),
+        this.db.getAllPullRequestsInRepository(repository),
+      ])
+      const records = new Array<IPullRequest>()
+      records.push(...upstreamRecords, ...forkRecords)
+      return records
```

Comment:
Could we get away with skipping the intermediate array creation here?
```suggestion
      return [...upstreamRecords, ...forkRecords]
```